events/test_routes.py

```diff
@@ -59,6 +60,14 @@ def test_add_invalid_event(self):
 
         self.assertEqual(self.coll.count_documents({}), 0)
 
+    def test_db_not_defined(self):
+        """Test adding event when DB connection is undefined."""
+        self.coll.insert_one = MagicMock(side_effect=DBNotConnectedError)
+        response = self.client.post('/v1/add', data=VALID_REQUEST_INFO)
```

Comment:
Nice test!

One other way to do this would be to patch the mongo URL to choose an invalid connection string: https://github.com/mongomock/mongomock/blob/develop/tests/test__mongomock.py#L54

But that might require restructuring your code.

Reply:
If you want to try this, maybe something like this would work.
```python
del os.environ["MONGODB_URI"]  # unset db config variable
app.config["COLLECTION"] = connect_to_mongodb()  # will throw DBNotConnectedError on access
response = self.client.post('/v1/add', data=VALID_REQUEST_INFO)  # will throw inside handler
```

Reply:
Seems to work, thanks! I just need to make sure to replace the MONGODB_URI variable afterwards otherwise later tests will fail.

Reply:
Nevermind, seems to work when testing on Travis - it doesn't seem to want me to mess with env variables. I think I'll just stick to the previous method.

Reply:
Here's an example of using a context manager to set environment variables (use the last comment in the gist): https://gist.github.com/igniteflow/7267431

Reply:
The error in Travis was because `del os.environ["MONGODB_URI"]` fails if `"MONGODB_URI"` is not a key in the environment. Either way you choose to do it, you might have to check if `os.environ["MONGODB_URI"]` is set first.

Reply:
I think the `contextlib` approach should be able to work around this:

```from contextlib import contextmanager

@contextmanager
def environ(env):
    """Temporarily set environment variables inside the context manager and
    fully restore previous environment afterwards
    """
    original_env = {key: os.getenv(key) for key in env}
    os.environ.update(env)
    try:
        yield
    finally:
        for key, value in original_env.items():
            if value is None:
                del os.environ[key]  # You could set the value to "WHAT, see line XYZ" instead of del
            else:
                os.environ[key] = value
```

Reply:
Implemented with context manager in 6543e0e (with fix in 64c409b)